lucene/core/src/java/org/apache/lucene/search/MemoryAccountingBitsetCollector.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.misc.CollectorMemoryTracker;
+import org.apache.lucene.util.FixedBitSet;
+
+/** Bitset collector which supports memory tracking */
+public class MemoryAccountingBitsetCollector extends SimpleCollector {
+
+  final CollectorMemoryTracker tracker;
+  FixedBitSet bitSet = new FixedBitSet(0);
+  int length = 0;
+  int docBase = 0;
+
+  MemoryAccountingBitsetCollector(CollectorMemoryTracker tracker) {
```

Comment:
do we expect users to call this constructor themselves? If yet it should be public otherwise they won't be able to create it?